hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeUpdateHandler.java

```diff
@@ -36,47 +48,81 @@
  */
 @Singleton
 public class NodeUpdateHandler implements TransactionHandler {
+    private final AddressBookValidator addressBookValidator;
 
     @Inject
-    public NodeUpdateHandler() {
-        // Exists for injection
+    public NodeUpdateHandler(@NonNull final AddressBookValidator addressBookValidator) {
+        this.addressBookValidator =
+                requireNonNull(addressBookValidator, "The supplied argument 'addressBookValidator' must not be null");
     }
 
     @Override
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
         final var op = txn.nodeUpdate();
-        throw new UnsupportedOperationException("need implementation");
+        validateFalsePreCheck(op.nodeId() < 0, INVALID_NODE_ID);
+        if (op.hasAccountId()) {
+            final var accountId = validateAccountID(op.accountIdOrElse(AccountID.DEFAULT), INVALID_NODE_ACCOUNT_ID);
+            validateFalsePreCheck(!accountId.hasAccountNum() && accountId.hasAlias(), INVALID_NODE_ACCOUNT_ID);
+        }
+        if (op.hasGossipCaCertificate())
+            validateFalsePreCheck(op.gossipCaCertificate().equals(Bytes.EMPTY), INVALID_GOSSIP_CAE_CERTIFICATE);
```

Comment:
We should extract same validations happening in `NodeCreateHandler` and `NodeUpdateHandler` to base handler to avoid duplication ?

Reply:
For the update, we do the check if the field is present. So they are not exactly same as create. validateXXX is same.